frontend/model/contracts/group.js

```diff
@@ -1220,14 +1220,19 @@ sbp('chelonia/defineContract', {
     'gi.contracts/group/leaveChatRoom': {
       validate: actionRequireActiveMember(objectOf({
         chatRoomID: stringMax(MAX_HASH_LEN, 'chatRoomID'),
-        memberID: optional(stringMax(MAX_HASH_LEN), 'memberID')
+        memberID: optional(stringMax(MAX_HASH_LEN), 'memberID'),
+        // `joinedHeight` is the height used in the corresponding join action
+        joinedHeight: numberRange(1, Number.MAX_SAFE_INTEGER)
```

Comment:
So... in this action we can expect `joinedHeight` to be in `data`, but below where it says:

```js
state.chatRooms[data.chatRoomID].members[memberID].joinedHeight !== data.joinedHeight
```

(And similar)

Is this a problem for existing clients who did not set `joinedHeight` because they used an older version of `'gi.contracts/group/joinChatRoom'`?

Reply:
Hm, I think it'd indeed be an issue. A way to possibly fix this could be to test for presence of this attribute (also, it won't be in `data` for old contracts). Those contracts may have issues with reference counts without this information.

Reply:
Another way would be to implement contract upgrades. In this case, we'd set joinedHeight to the height of the update.